controllers/awsmachine_controller.go

```diff
@@ -137,6 +137,8 @@ func (r *AWSMachineReconciler) getObjectStoreService(scope scope.S3Scope) servic
 // +kubebuilder:rbac:groups="",resources=secrets;,verbs=get;list;watch
 // +kubebuilder:rbac:groups="",resources=namespaces,verbs=get;list;watch
 // +kubebuilder:rbac:groups="",resources=events,verbs=get;list;watch;create;update;patch
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=kubemarkmachinetemplates,verbs=get;list;watch;create;update;patch;delete
```

Comment:
Why do we need these permissions?

Reply:
Yeah, this is a requirement for the autoscaler indeed. I guess we can do that, but that just adds an extra step for the user which they will need to automate instead of us doing this for them which results in a better UX IMHO.

Reply:
I think this should be `awsmachinetemplates` instead of `kubemarkmachinetemplates`

Reply:
Hah, that's actually a very good point. :D

Reply:
Pretty sure it is `cluster-autoscaler` that needs this permission, not `cluster-api-provider-aws`, so don't think this is needed here (unless I'm misunderstanding what this does)?

Reply:
If these permissions are not needed here, then after this is removed, LGTM, we can even try to add this feature to the upcoming v1beta1 release as the changes are not big.

Reply:
Sounds good. I'll remove it!

Reply:
So, the reason why I added this is that the reference implementation was also adding these RBAC rules and the proposal states that the service account associated with the cluster-autoscaler must have these permissions.

I'll update the document that this now something that the user has to do.

Reply:
Ah! I think we implicitly already allow this through various `+kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io` rbacs. :D That's why it was working regardless. Anyways, I updated the docs and the code. Thanks you two. :)